### PR TITLE
refactor: move iam-specific functionality to a shared class 

### PR DESCRIPTION
**Pull request by dpopp07**
This will allow other token managers that interact with the IAM service to share configuration and functionality. This change is being made in advance of a new container-based token manager being added that will utilize this shared functionality, as it will interact with the IAM service as well.

cc @pyrooka

### CODE REVIEW

File: test/unit/iam-token-manager.test.js [hunk 1]
```diff
@@ -119,294 +141,5 @@ describe('iam_token_manager_v1', () => {
     const token = await instance.getToken();
     expect(token).toBe(ACCESS_TOKEN);
     expect(requestMock).not.toHaveBeenCalled();
-    done();
-  });
-
```
Comment (dpopp07) [inline test/unit/iam-token-manager.test.js:124]: All of these tests below were consolidated and are covered by the tests for the new IAM request based token manager

File: test/unit/logger.test.js [hunk 2]
```diff
@@ -17,11 +17,11 @@ describe('Logger', () => {
...
-    expect(logger.warn.enabled).toBe(false);
-    expect(logger.verbose.enabled).toBe(false);
-    expect(logger.info.enabled).toBe(false);
-    expect(logger.error.enabled).toBe(false);
-    expect(logger.debug.enabled).toBe(false);
+    expect(logger.warn.enabled).toBeFalsy();
+    expect(logger.verbose.enabled).toBeFalsy();
+    expect(logger.info.enabled).toBeFalsy();
+    expect(logger.error.enabled).toBeFalsy();
+    expect(logger.debug.enabled).toBeFalsy();
```
Comment (dpopp07) [inline test/unit/logger.test.js:24]: These were failing locally. Honestly, I'm not sure how they were ever passing. We never set those parameters to `false`

**Review by padamstx — APPROVED** [review]
LGTM

**Pull request merged**